docs/getting-started.asciidoc

```diff
@@ -1,30 +1,26 @@
 [[getting-started]]
+== Introduction
 
-ifdef::env-github[]
-NOTE: For the best reading experience,
-please view this documentation at https://www.elastic.co/guide/en/apm/agent/python/current/getting-started.html[elastic.co]
-endif::[]
+The Elastic APM Python agent sends performance metrics and error logs to the APM Server.
+It has built-in support for Django and Flask performance metrics and error logging, as well as generic support of other WSGI frameworks for error logging.
 
-== Getting started
+[float]
+[[how-it-works]]
+=== How does the Agent work?
 
-Welcome to the APM Python agent docs.
+The agent auto-instruments <<supported-technologies,supported technologies>> and records interesting events,
+like HTTP requests and database queries. To do this, it {CONTENT_HERE}
+This means that for the supported technologies, there are no code changes required.
 
-The Elastic APM Python agent sends performance metrics and error logs to the APM Server.
-It has built-in support for Django and Flask performance metrics and error logging, as well as generic support of other WSGI frameworks for error logging.
+The Agent automatically {CONTENT_HERE}
+
+These events, called Transactions and Spans, are sent to the APM Server.
+The APM Server converts them to a format suitable for Elasticsearch, and sends them to an Elasticsearch cluster.
+You can then use the APM app in Kibana to gain insight into latency issues and error culprits within your application.
```

Comment:
One of the things this PR aims to do is add in a section on how the agent does instrumentation. I've added some boilerplate text here to hopefully assist with this. There are two {CONTENT_HERE} sections above that I'll need one of you to fill out. Feel free to write something different or amend this as necessary.

I've also updated https://github.com/elastic/apm/issues/146 with links to examples in other Agent's docs.